test/trip_dispatch_test.rb

```diff
@@ -120,6 +122,37 @@ def build_test_dispatcher
           expect(trip.driver.trips).must_include trip
         end
       end
-    end
-  end
+    end 
+
+  
+  
+     describe "request trip" do
```

Comment:
These tests look good! It looks like you implemented some of the optional conditions for choosing a driver. It is important to test these as well.